api/src/main/java/org/openmrs/CohortMembership.java

```diff
@@ -101,4 +115,16 @@ public void setEndDate(Date endDate) {
 	public int compareTo(CohortMembership o) {
 		return this.getPatient().getPatientId() - o.getPatient().getPatientId();
 	}
+	
+	@Override
```

Comment:
Codacy is correct here, you always have to implement hashCode when you implement eqauls and they should be consistent with each other, 2 objects that are equal should always have the same hashcode